msdk-id-sirius/src/main/java/SiriusIdentificationMethod.java

```diff
@@ -0,0 +1,349 @@
+      int k = 0;
+
+      // Add items from `range` into array with default elements
+      for (IIsotope isotope: range.isotopes()) {
+        int atomicNumber = isotope.getAtomicNumber();
+        final Element element = periodicTable.get(atomicNumber);
+        elements[defaultElements.length + k++] = element;
+      }
+
+      // Generate initial constraint w/o concrete Element range
+      FormulaConstraints constraints = new FormulaConstraints(new ChemicalAlphabet(elements));
+
+      // Specify each Element range
+      for (IIsotope isotope: range.isotopes()) {
+        int atomicNumber = isotope.getAtomicNumber();
+        final Element element = periodicTable.get(atomicNumber);
+        int min = range.getIsotopeCountMin(isotope);
+        int max = range.getIsotopeCountMax(isotope);
+
+        constraints.setLowerbound(element, min);
+        if (max!= maxNumberOfOneElements) constraints.setUpperbound(element, max);
+      }
+
+      logger.debug("ConstraintsGenerator finished");
+      return constraints;
+    }
+
+  }
+
+  public double getParentMass() {
+    return parentMass;
+  }
+
+  public IonType getIonization() {
+    return ion;
+  }
+
+  public List<MsSpectrum> getMsSpectra() {
+    return ms1;
+  }
+
+  public List<MsSpectrum> getMs2Spectra() {
+    return ms2;
+  }
+
+  public int getNumberOfCandidates() {
+    return numberOfCandidates;
+  }
+
+  /**
+   *
+   * <p> Method for reading custom MS files (just 2 columns with mz & intentsity values </p>
+   * Does similar to mgf parser functionality
+   */
+  public static MsSpectrum readCustomMsFile(File file, String delimeter)
+      throws IOException, MSDKRuntimeException {
+    logger.info("Started reading {}", file.getName());
+
+    Scanner sc = new Scanner(file);
+    ArrayList<String> strings = new ArrayList<>();
+    while (sc.hasNext()) {
+      strings.add(sc.nextLine());
+    }
+    sc.close();
+
+    int size = strings.size();
+    double mz[] = new double[size];
+    float intensity[] = new float[size];
+
+    int index = 0;
+    for (String s : strings) {
+      String[] splitted = s.split(delimeter);
+      if (splitted.length == 2) {
+        mz[index] = Double.parseDouble(splitted[0]);
+        intensity[index++] = Float.parseFloat(splitted[1]);
+      } else {
+        throw new MSDKRuntimeException("Incorrect spectrum structure");
+      }
+    }
+
+    MsSpectrumType type = SpectrumTypeDetectionAlgorithm.detectSpectrumType(mz, intensity, size);
+    logger.info("Finished reading {}", file.getName());
+
+    return new SimpleMsSpectrum(mz, intensity, size, type);
+  }
+
+  /**
+   * <p> Method for processing spectra by Sirius module </p>
+   * Transformation of MSDK data structures into Sirius structures and processing by Sirius
+   * Method is left to be protected for test coverage
+   */
+  protected List<IdentificationResult> siriusProcessSpectra() throws MSDKException {
+    Spectrum<Peak> siriusMs2;
+    MsSpectrum msdkSpectrumMs2 = ms2.remove(0);
```

Comment:
This looks strange - why are you modifying the contents of ms2?

Reply:
I think you should not modify the original ms2 list here. Because the user who is using the module might not expect that the ms2 list (which the user gave to you as a parameter), will be modified during the process. It is better if you create a copy of the list for your own use, then you can do anything you want with it (the copy of the list will contain the references to the same spectra as the original list, there is no need to create a copy of each spectrum).

Reply:
Oh, really, thank you for advice